test/unit/disposable_common.dart

```diff
@@ -352,6 +351,89 @@ void testCommonDisposable(Func<StubDisposable> disposableFactory) {
     });
 
     test(
+        'should un-manage subscription when controller is closed '
+        'due to an error and cancelOnError is true', () async {
+      var previousTreeSize = disposable.disposalTreeSize;
+
+      // ignore: close_sinks
+      var controller = new StreamController<Null>();
+      // ignore: cancel_subscriptions
+      var subscription = disposable.listenToStream(controller.stream, (_) {},
+          cancelOnError: true, onError: (_, [__]) {});
+      expect(disposable.disposalTreeSize, equals(previousTreeSize + 1));
+
+      controller.addError(new Exception('intentional'));
+      await new Future(() {});
+
+      expect(disposable.isDisposed, isFalse);
+      expect(disposable.disposalTreeSize, equals(previousTreeSize));
+    });
+
+    test(
+        'should not un-manage subscription when controller is closed '
+        'due to an error and cancelOnError is false', () async {
```

Comment:
I think this test desc could be clarified, the controller isn't being closed here because `cancelOnError` is false.